ui/app/AppLayouts/Wallet/popups/BuyCryptoModal.qml

```diff
@@ -0,0 +1,97 @@
+import QtQuick 2.14
+import QtQuick.Layouts 1.0
+import QtQml.Models 2.14
+import SortFilterProxyModel 0.2
+
+import StatusQ.Popups.Dialog 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ 0.1
+
+import utils 1.0
+
+StatusDialog {
+    id: root
+
+    property var onRampProvidersModel
+
+    padding: Style.current.xlPadding
+    implicitWidth: 560
+    title: qsTr("Buy assets")
+
+    ColumnLayout {
+        anchors.fill: parent
+        spacing: 20
+
+        StatusSwitchTabBar {
+            id: tabBar
+            objectName: "tabBar"
+            Layout.alignment: Qt.AlignHCenter
+            Layout.fillWidth: true
+            StatusSwitchTabButton {
+                text: qsTr("One time")
+            }
+            StatusSwitchTabButton {
+                text: qsTr("Recurrent")
+            }
+        }
+
+        StatusListView {
+            id: providersList
+            objectName: "providersList"
+            Layout.fillWidth: true
+            Layout.preferredHeight: root.onRampProvidersModel.count * currentItem.height
```

Comment:
Ive changed it to rowCount now!

Reply:
I think I'm missing something. Why wouldn't `Layout.fillHeght: true` work? 🤔 Or `Layout.preferredHeight: contentHeight` (not a great option, but still an option😄 )

Reply:
> But that one doesn't notify of changes!

@caybro this list is static and comes from the go side, should not change expect for a new release so that not an issue.

Reply:
> I think I'm missing something. Why wouldn't `Layout.fillHeght: true` work? 🤔 Or `Layout.preferredHeight: contentHeight` (not a great option, but still an option😄 )

@alexjba very good point! I dont like this very much too, but Ive had to do it because I need the same height for both the tabs and the "recurrent" tab may have 0 or a subset of the items in the first tab. This helps achieve that result. see this design https://www.figma.com/design/TS0eQX9dAZXqZtELiwKIoK/Swap---Milestone-1?node-id=4156-81099&m=dev

Reply:
You can set the `height` on the delegate instead, and use the same for the header/tabbar

Reply:
thats not what I mean, I meant that there are two tabs "One Time" and "Recurrent". One maybe 3 items the other only 1 or 0.
I want the height for both to be the same like in the image below -
<img width="424" alt="Screenshot 2024-05-30 at 10 12 46 AM" src="https://github.com/status-im/status-desktop/assets/60327365/7969a47e-6107-4ed9-a338-f5592cabbc2d">

Reply:
Ive given the modal a fixed height instead now. @caybro @alexjba  please check?

Reply:
ohh, this is great! Thanks for taking the time to look into it! 🍻